static/js/index.js

```diff
@@ -0,0 +1,147 @@
+    for (i in logs) {
+        pNode = document.createElement('p')
+        pNode.innerText = logs[i]
+        logsNode.appendChild(pNode)
+    }
+}
+
+const setRadioButtons = (scraperSettings) => {
+    const loopModeNode = document.querySelectorAll('input[name="loop-radio-group"]')
+    const mediaModeNode = document.querySelectorAll('input[name="media-radio-group"]')
+    const idsModeNode = document.querySelectorAll('input[name="ids-radio-group"]')
+    loopModeNode.forEach( (el) => {
+        if (el.value === scraperSettings['loop_mode'].toString()) {
+            el.checked = true
+        }
+    })
+
+    mediaModeNode.forEach((el) => {
+        if (el.value === scraperSettings['media_mode'].toString()) {
+            el.checked = true
+        }
+    })
+
+    idsModeNode.forEach((el) => {
+        if (el.value === scraperSettings['ids_source'].toString()) {
+            el.checked = true
+        }
+    })
+}
+
+const checkShutdown = async () => {
+    const timeInterval = 2000
+    const makeRequest = async () => {
+        const requestUrl = `${baseUrl}/${API_PREFIX}/status/`;
+        const responseData = await (await fetch(requestUrl)).json()
+        return responseData
+    }
+
+    const updatePage = async () => {
+        try {
+            const response = await makeRequest()
+            if (response['status'] === "stopped") {
+                location.reload()
+            } else {
+                setTimeout(updatePage, timeInterval)
+            }
+         } catch {
+             location.reload()
+         }
+    }
+
+    setTimeout(updatePage, timeInterval)
+}
+
+const updateCommandButton = (buttonStatus) => {
+    const commandButton = document.getElementById('command-button')
+    if (buttonStatus === "running") {
+        commandButton.disabled = false
+        commandButton.classList = "btn stop-btn"
+        commandButton.innerText = "STOP"
+        commandButton.value = "stop"
+    } else if (buttonStatus === "shutdown") {
+        commandButton.disabled = true
+        commandButton.classList = "btn update-btn"
+        commandButton.innerText = "updating..."
+        checkShutdown()
+    } else {
+        commandButton.disabled = false
+        commandButton.classList = "btn start-btn"
+        commandButton.innerText = "START"
+        commandButton.value = "start"
+    }
+}
+
+const getScraperStatus = async () => {
+    const requestUrl = `${baseUrl}/${API_PREFIX}/index/`;
+    const responseData =  await (await fetch(requestUrl)).json()
+    const outputNode = document.getElementById('scraping-results')
+    setScrapingLogs(responseData['log_lines'])
+    setRadioButtons(responseData['scraper_settings'])
+    updateCommandButton(responseData['scraper_status'])
+    outputNode.innerText = `${responseData['output']['scraped_media'] ?? 0} media scraped`
+}
+
+const startScraping = async () => {
+    const userLimit = Number(document.getElementById('user-limit').value)
```

Comment:
Can't you set type="number" in the html and be done with the conversion?

Reply:
If i set type="number", the typeof(input.value) is string, so I have to cast it anyway